src/com/ubcsolar/sim/SimController.java

```diff
@@ -115,6 +115,110 @@ public void runSimulation(int laps, long startTime)
+		//idea is to use runSimV2 for every part of the route where the car travels at constant speed until the speed change
+		//the point with the different speed would be the last point, and the point after would be the start of the next simulation
+		GeoCoord start = routeToTraverse.getTrailMarkers().get(1); //first point of the simulation (need to ask why its 1, running point 0 to 1 is always a fail)
+		GeoCoord end; //last point of the simulation
+		TreeMap<Integer, ForecastIO> inflectionPoints = mySession.getMyWeatherController()
+				.findInflectionPoints(routeToTraverse, currentForecastReport.getForecasts());
+		ForecastIO current_forecast = inflectionPoints.get(inflectionPoints.keySet().toArray()[0]);
+		long nextStartTime = startTime;
+		long totalTime = 0;
+		
+		for (int i = 1; i <= laps; i++) {
+			//iterate through all the points except for the last point (perserve last point for one last runSimV2 call since car may be traveling at constant speeds
+			//until the end
+			for (int point_index = 1; point_index < routeToTraverse.getTrailMarkers().size(); point_index++) {
+				//check to see if an inflection point has reached and a new forecast is used
+				if (inflectionPoints.keySet().contains(i)) {
+					current_forecast = inflectionPoints.get(i);
+				}
+				
+				//check to see if there is a speed change
+				//if there is, run a simulation and update list of sim frames, new start speed, new start point, start time and car status using results
+				if (Math.abs(manualSpeedProfile.get(start).get(i) - manualSpeedProfile.get(routeToTraverse.getTrailMarkers().get(point_index)).get(i)) > 0.000001 ||
+						point_index == routeToTraverse.getTrailMarkers().size()) {
+					end = routeToTraverse.getTrailMarkers().get(point_index);
+					//min charge is set to 0. Since speeds are decided by users and no adjustments will be made, we let the simulation use up all the charge
+					//for each runSimV2 call. We assume that if the cannot make it through without conserving charge for the next runSimV2 call (giving a non-zero
+					//value for min charge), it will not make it through if charge is conserved (may need to change for single lap routes, definitely needs tweaking for
+					//for multiple laps)
+					result = new SimEngine().runSimV2(routeToTraverse, start, end,
+							lastCarReported, manualSpeedProfile.get(start).get(i), manualSpeedProfile.get(end).get(i),
+							nextStartTime, i, 0, current_forecast);
+					//if sim is not successful, remove all the points before the end of the sim (only want to graph to the point the car fails, so speeds after 
+					//that needs to be removed in speed profile)
+					if(!result.wasRunSuccessful()) {
+						System.err.println("Car cannot make it through the route with selected speeds");
+						for (int removed_point = routeToTraverse.getTrailMarkers().size() - 1; (!end.equals(removed_point) && removed_point >= 0);removed_point--) {
```

Comment:
it checks for the last point that was the parameter for the failed runSimV2() call. The loop removes all the points after the failed call in the speedprofile, so the graph shows speeds until it runs out of charge.